Src/VimCore/EditorUtil.fs

```diff
@@ -1782,6 +1782,43 @@ module TextViewUtil =
         let caret = GetCaret textView
         caret.EnsureVisible()
 
+    /// Check whether we should recenter the view on point
+    let CheckRecenterOnPoint (textView : ITextView) point =
+
+        // Be careful because using ITextViewLines can result in an exception.
+        try
+            let textViewLine = textView.GetTextViewLineContainingBufferPosition point
+            let onScreen = textViewLine.VisibilityState = Formatting.VisibilityState.FullyVisible
+            if not onScreen then
+
+                // The line the point is on is not on screen.
+                // We handle three cases.  The point line:
+                // 1) precedes the first visible line by less than half a screen
+                // 2) follows the last visible line by less than half a screen
+                // 3) is further away than half a screen
+                // Vim uses pseudo-scrolling for cases 1) and 2) and centers
+                // the point vertically within the screen for 3).
+                let pointLine = SnapshotPointUtil.GetContainingLine point
+                let pointLineNumber = pointLine.LineNumber
+                let textViewLines = textView.TextViewLines

```

Comment:
The `ITextView.TextViewLines` property can also return `null` in some cases. `TextViewUtil.GetTextViewLines` tries to encapsulate this, and the exception oddity, in an option.

Reply:
I've only recently understood the advantage of the utility functions, for example for currying, for syntax, and for cases like this where an option result is a better match for F#. I'll change this code to use the method you describe.